pkg/controller/worker/helper.go

```diff
@@ -59,3 +63,21 @@ func (w *workerDelegate) updateWorkerProviderStatus(ctx context.Context, workerS
 		return nil
 	})
 }
+
+func makeCopyOfMap(mp map[string]string) map[string]string {
+	var mpDeepCopy = make(map[string]string)
+	for k, v := range mp {
+		mpDeepCopy[k] = v
+	}
+	return mpDeepCopy
+}
+
+func addAwsCsiDriverTopologyLabel(poolLabels map[string]string, zone string) map[string]string {
+	if _, exists := poolLabels[AwsCsiDriverTopologyKey]; !exists {
+		var poolLabelsDeepCopy = makeCopyOfMap(poolLabels)
+		poolLabelsDeepCopy[AwsCsiDriverTopologyKey] = zone
+		return poolLabelsDeepCopy
+	}
+
+	return poolLabels
+}
```

Comment:
Why do we need a copy?

Reply:
if I made the changes to pool.Labels directly , I would have to call another helper function to revert those changes as pool.Labels might be used somewhere else
I thought its better to make a copy in that case, even though copying will add extra time complexity

Reply:
> I would have to call another helper function to revert those changes as pool.Labels might be used somewhere else

I can't follow, can you explain again? Why do you need to revert the changes again? 😕

Reply:
The pool.Labels are generic to that worker which might be hosting multiple zones. Hence he is making a copy to create one specific for the machineDeployment for a `zone`.

Reply:
Hm, I see. You might want to use https://github.com/gardener/gardener/blob/b72ac4af0c9d00454be8b75eba00d7eaf8df8b48/pkg/utils/miscellaneous.go#L70-L91 to save some code, i.e.

```golang
				Labels:               utils.MergeMaps(pool.Labels, map[string]string{AwsCsiDriverTopologyKey: zone}),
```

WDYT?

Reply:
I was wondering if there is already util for this earlier. However, I overlooked it. Good point, we can use this. WDYT @himanshu?

Reply:
yeah thanks @rfranzke , it will save writing  helper functions, I will use it